### ISSUE TITLE
Deleting a user does not delete its player and they become non-deletable

### ISSUE THREAD

**Issue by johanfleury** [deluan/navidrome#345]
As an admin, I removed a user through the dashboard and now I can’t remove its registered players.

When I try to remove the players I get the “players removed [undo]” message, but they reappear a few seconds latter.

I had to remove the players directly from the DB: `DELETE FROM player WHERE name LIKE '%[REDACTED USERNAME]%';`.

Having a look at the `player` table, it seems that you’re storing user names rather than a reference to the `user` table which would allow using SQL’s `ON DELETE CASCADE` constraint:

```
CREATE TABLE player
(
    id varchar(255) not null primary key,
    name varchar not null,
    type varchar,
    user_name varchar not null,   <-- Could be a reference to `user.id`
    client varchar not null,
    ip_address varchar,
    last_seen timestamp,
    max_bit_rate int default 0,
    transcoding_id varchar,
        unique (name),
        foreign key (transcoding_id)
           references transcoding(id)
                  on update restrict
                  on delete restrict
);
```

**Comment by deluan** [comment]
Thanks for the report. 

Yes, there's very little referential integrity on Navidrome's DB. This was a conscious decision to simplify scanning, but should not be the case for non-music related tables (users, players, transcodings, etc...)

I'll fix this and may introduce the appropriated foreign keys, but not sure if I'll use `ON DELETE CASCADE`, as AFAIK it is not ANSI SQL

**Comment by deluan** [comment]
Hey @johanfleury, any suggestions about what to do with playlists owned by the deleted user? The options I see: 
- Cascade delete them (I think this is the best)
- Add the ability to the UI to re-assign a playlist (change the owner)

Let me know your thoughts

**Comment by johanfleury** [comment]
I’m no UX expert, but I would separate the two features:

* On user deletion, the operator should be warned that their action will remove everything associated to the user (preferences, playlists, players, transcoding options, …) then, unpon validation, everything sould be deleted in cascade.
* At the same time, we could add the ability to reassign playlists either in the playlists view or as a action in the user list.

I can probably work on user deletion and database migration and send you a PR if you’d like.

**Comment by deluan** [comment]
Yeah, that makes sense. I already started working on the first item in in PR #346 (thanks for the offer anyway), and may add the second item as a future enhancement.
